src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs

```diff
@@ -132,7 +132,7 @@ private void OnResolve(Type serviceType, IServiceScope scope)
 
         private void ValidateService(ServiceDescriptor descriptor)
         {
-            if (descriptor.ServiceType.IsGenericType && !descriptor.ServiceType.IsConstructedGenericType)
```

Comment:
>IsConstructedGenericType [](http://example.com/codeflow?start=80&length=24)

@maryamariyan could you explain more about this change? 

From the code I think when having IsConstructedGenericType should try to Get or create the callsite? for IEnumerable<>, maybe need to check for interfaces instead?

Reply:
Adding a comment to the code might be a good idea.

Reply:
What I noticed is if the type is generic (constrained or not) `ValidateService` would incorrectly resolve the generics.

Also since we already have the test below expecting `ValidateOnBuild` to be skipped for generics, then I'm more convinced this is the fix we want for the reported bug.

https://github.com/dotnet/runtime/blob/8ea879af03852e3d7d916badcf836af61fabb33c/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderValidationTests.cs#L137-L146

I think the reason to skip `ValidateOnBuild` should be that there is no knowledge upon build for what `T` user would use later for any given `XGeneric<T>` service. So would not make sense to validate generic service on build.

Added comment: 
```cs
// Skip validation for generic service types
```

Reply:
This is a breaking change right? Now closed generics won't be validated?